generic3g/specs/VerticalDimSpec.F90

```diff
@@ -43,12 +48,43 @@ module mapl3g_VerticalDimSpec
       
 contains
 
-
    elemental logical function equal_to(a, b)
       type(VerticalDimSpec), intent(in) :: a, b
       equal_to = a%id == b%id
    end function equal_to
 
+   subroutine write_formatted(this, unit, iotype, v_list, iostat, iomsg)
+      class(VerticalDimSpec), intent(in) :: this
+      integer, intent(in) :: unit
+      character(*), intent(in) :: iotype
+      integer, intent(in) :: v_list(:)
+      integer, intent(out) :: iostat
+      character(*), intent(inout) :: iomsg
+
+      integer :: id
+      character(len=ESMF_MAXSTR) :: dim_spec_str
+
+      id = this%id
+      select case(id)
+      case(-1)
+         dim_spec_str = "VERTICAL_DIM_UNKNOWN"
+      case(1)
+         dim_spec_str = "VERTICAL_DIM_NONE"
+      case(2)
+         dim_spec_str = "VERTICAL_DIM_CENTER"
+      case(3)
+         dim_spec_str = "VERTICAL_DIM_EDGE"
+      case(4)
+         dim_spec_str = "VERTICAL_DIM_MIRROR"
+      ! case default
+      !    _FAIL("Invalid vertical dim spec")
+      end select
```

Comment:
I know it is tedious, but we probably should relocate these strings as parameters next to where the various spec parameters are defined.   Will make it much easier to maintain if we ever need to extend.

Reply:
Can be a separate PR, but ... don't forget.

Reply:
Created new issue, just so I don't forget.